### PR TITLE
Bugfix/objectconfig inheritance improvements

### PR DESCRIPTION
**Pull request by abolder**
Improved inheritance of the dlmsprofiles within the ObjectConfigService. So selectable objects will override existing inherited selectable objects.

### CODE REVIEW

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java [hunk 1]
```diff
@@ -38,54 +40,61 @@ void setUp() throws IOException, ObjectConfigException {
   }
 
   @Test
-  void testGetCosemObjects() throws ObjectConfigException {
+  void getCosemObjectIsNull() throws ObjectConfigException {
     final String protocolName = "SMR";
-    final String protocolVersion50 = "5.0.0";
-    final String protocolVersion51 = "5.1";
-    final String protocolVersion52 = "5.2";
+    final String protocolVersion50 = "4.3";
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Naam of value aanpassen

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java [hunk 2]
```diff
@@ -38,54 +40,61 @@ void setUp() throws IOException, ObjectConfigException {
...
-        });
+        ObjectConfigException.class, () -> this.objectConfigService.getCosemObjects("ABC", "12"));
   }
 
   @Test
-  void testGetCosemObject() throws ObjectConfigException {
-    final String protocolVersion50 = "5.0.0";
+  void getCosemObjectNotFoundShouldThrowIllegalArgumentException() {
+    final String protocolVersion50 = "4.3";
     final String protocolName = "SMR";
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Naam of value aanpassen

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java [hunk 3]
```diff
@@ -38,54 +40,61 @@ void setUp() throws IOException, ObjectConfigException {
...
-              protocolName, protocolVersion50, DlmsObjectType.AVERAGE_CURRENT_L1);
-        });
+        () ->
+            this.objectConfigService.getCosemObject(
+                protocolName, protocolVersion50, DlmsObjectType.MBUS_DIAGNOSTIC));
+  }
+
+  @Test
+  void getCosemObjectShouldFindObject() throws ObjectConfigException {
+    final String protocolVersion50 = "4.3";
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Naam value

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java (hunk 1 above)
Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Yes, aangepast.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java (hunk 2 above)
Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Aangepast

File: osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java (hunk 3 above)
Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/test/java/org/opensmartgridplatform/dlms/services/ObjectConfigServiceTest.java]: Aangepast

**Review by kroesctrl — APPROVED** [review]

**Comment by abolder** [comment]
✔︎ GXF: https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/65/

**Pull request merged**